flink-core/src/main/java/org/apache/flink/configuration/CoreOptions.java

```diff
@@ -527,23 +527,41 @@ public static String[] mergeListsToArray(List<String> base, List<String> append)
      * open. Unlimited be default.
      */
     public static ConfigOption<Integer> fileSystemConnectionLimit(String scheme) {
-        return ConfigOptions.key("fs." + scheme + ".limit.total").intType().defaultValue(-1);
+        return ConfigOptions.key("fs." + scheme + ".limit.total")
+                .intType()
+                .defaultValue(-1)
+                .withDescription(
+                        "The connection limit for the file system "
+                                + scheme
+                                + ". The valid value must >= -1. Values of -1 or 0 mean there is no connection limit.");
     }
 
     /**
      * The total number of input connections that a file system for the given scheme may open.
      * Unlimited be default.
      */
     public static ConfigOption<Integer> fileSystemConnectionLimitIn(String scheme) {
-        return ConfigOptions.key("fs." + scheme + ".limit.input").intType().defaultValue(-1);
+        return ConfigOptions.key("fs." + scheme + ".limit.input")
```

Comment:
nit: as the 3 configurations only differ with the key and first word(s) of the description, could we define a method with the common code. Maybe a common method with connectionLimitType. 
send in "total" "input" or "output" as connectionLimitType
key would have 
.limit. + connectionLimitType
and the description could start
"The " +  connectionLimitType + " connection limit ....

WDYT?

Reply:
This is a good idea. But I'm not sure if it is worth!